### ISSUE TITLE
azurerm_redis_cache major version update failed from version 4 to version 6

### ISSUE THREAD

**Issue by pnjha** [hashicorp/terraform-provider-azurerm#18058]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.15.5

### AzureRM Provider Version

2.85.0

### Affected Resource(s)/Data Source(s)

azurerm_redis_cache

### Terraform Configuration Files

```hcl
{
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "hashicorp/azurerm"
      }
    }
  },
  "provider": {
    "azurerm": {
      "features": {},
      "client_id": "<redacted>",
      "client_secret": "<redacted>",
      "subscription_id": "<redacted>",
      "tenant_id": "<redacted>"
    }
  },
  "resource": {
    "azurerm_redis_cache": {
      "redis_cache": {
        "name": "redis-5fac3126-e20d-43e3-b767-deca1dca0ef3",
        "location": "eastus",
        "resource_group_name": "<redacted>",
        "redis_version": "6",
        "capacity": 2,
        "family": "P",
        "sku_name": "Premium",
        "enable_non_ssl_port": false,
        "minimum_tls_version": "1.2",
        "redis_configuration": {
          "maxmemory_policy": "noeviction",
          "notify_keyspace_events": ""
        },
        "patch_schedule": {
          "day_of_week": "Sunday",
          "start_hour_utc": 4,
          "maintenance_window": "PT5H"
        },
        "shard_count": 1,
        "subnet_id": "<redacted>",
        "zones": ["1", "2", "3"],
        "tags": {},
        "timeouts": {
          "create": "8h",
          "update": "8h",
          "delete": "8h"
        },
        "lifecycle": {
          "prevent_destroy": true
        }
      }
    }
  },
  "output": {
    "id": {
      "value": "${azurerm_redis_cache.redis_cache.id}",
      "description": "Resource id for redis."
    },
    "hostname": {
      "value": "${azurerm_redis_cache.redis_cache.hostname}",
      "description": "Connection endpoint address for redis."
    },
    "port": {
      "value": "${azurerm_redis_cache.redis_cache.ssl_port}",
      "description": "Redis listener port."
    },
    "primary_access_key": {
      "value": "${azurerm_redis_cache.redis_cache.primary_access_key}",
      "description": "Redis primary access key.",
      "sensitive": true
    }
  }
}
```

### Debug Output/Panic Output

```shell
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  ~ update in-place
Terraform will perform the following actions:
  # azurerm_redis_cache.redis_cache will be updated in-place
  ~ resource "azurerm_redis_cache" "redis_cache" {
        id                            = "/subscriptions/<redacted>/resourceGroups/<redacted>/providers/Microsoft.Cache/Redis/redis-<redacted>"
        name                          = "redis-<redacted>"
      ~ redis_version                 = "4.0.14" -> "6"
        # (19 unchanged attributes hidden)
        # (3 unchanged blocks hidden)
    }
Plan: 0 to add, 1 to change, 0 to destroy.
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.
  Enter a value:
azurerm_redis_cache.redis_cache: Modifying... [id=/subscriptions/<redacted>/resourceGroups/<redacted>/providers/Microsoft.Cache/Redis/redis-<redacted>]
[GLOBAL] - error : Terraform operation failed with error: Command failed with exit code 1: yes yes | terraform -chdir=/usr/app/terraform_workspace apply

2022-07-30T11:54:31.438Z [INFO]  provider.terraform-provider-azurerm_v2.85.0_x5: preparing arguments for Azure ARM Redis Cache update.: timestamp=2022-07-30T11:54:31.438Z

Error: waiting for Redis Cache "redis-<redacted>" (Resource Group "<redacted>") to become available: unexpected state 'UpgradingRedisServerVersion', wanted target 'Succeeded'. last error: %!s(<nil>)
```

### Expected Behaviour

Terraform should have triggered the major version upgrade and then waited for the instance to become available once the upgrade is complete. 

### Actual Behaviour

Terraform triggers the update of major version from 4 to 6 but expects the instance state to be 'available' immediately which is wrong. It should have expected instance state to be 'UpgradingEngineVersion' as the instance state and should have waited for the instance to become available.

### Steps to Reproduce

1. Create an instance with version 4 
2. Change the version to 6 in the template
3. terraform apply

**Comment by rcskosir** [comment]
@pnjha Thank you for creating this issue. I attempted to reproduce it but I am no longer able to create new Azure Cache for Redis instances using Redis version 4: https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-retired-features#important-upgrade-timelines. Therefore I can not create a version 4 to then try and upgrade to version 6 to reproduce this issue. 
Have you been able to try this in a more recent provider version to see if this issue is still occurring?

**Comment by rcskosir** [comment]
Thanks again for opening this issue. Since time has passed and I was not able to reproduce this issue, I am going to mark this issue closed as “can’t reproduce”. If this is still an issue with the 3.x version of the provider please do let us know by opening a new issue, thanks!
